front/lib/api/files/upsert.ts

```diff
@@ -83,6 +84,78 @@ const upsertDocumentToDatasource: ProcessingFunction = async (
   return new Ok(undefined);
 };
 
+// Upload seachable document to dataSource
+// We expect the content of the file to be the JSON representation of a CoreAPIDataSourceDocumentSection.
+const upsertSectionDocumentToDatasource: ProcessingFunction = async (
+  auth,
+  { file, dataSource, upsertArgs }
+) => {
+  if (!upsertArgs || !("document_id" in upsertArgs)) {
+    return new Err({
+      name: "dust_error",
+      code: "invalid_request_error",
+      message: "Upsert args are required for section documents.",
+    });
+  }
+  if (upsertArgs.section) {
```

Comment:
We can also test and error if .text is defined?

Reply:
Updated, I'm now erroring if you send an upsertArgs